### ISSUE TITLE
Starscream 4.0.6 breaks GraphQLWebSocket protocol conformance to WebSocketDelegate

### ISSUE THREAD

**Issue by ynnadrules** [maticzav/swift-graphql#155]
**Describe the bug**
Starscream introduced a breaking change in the 4.0.5 [patch  release](https://github.com/daltoniam/Starscream/pull/812)

This package should pin Starscream's version to 4.0.4 or update `GraphQLWebSocket`'s conformance to Starscream's `WebSocketDelegate`.

**To Reproduce**

1. Update dependencies
2. Attempt to build
3. See build errors in GraphQLWebSocket:

**Expected behavior**
To build

**Screenshots**
<img width="322" alt="image" src="https://github.com/maticzav/swift-graphql/assets/159161/c80be3ba-e6bb-4985-92a2-0bd307197ea3">

**Desktop (please complete the following information):**

- OS: [e.g. iOS] iOS

**Smartphone (please complete the following information):**

- Device: [e.g. iPhone6] iPhone 14 Pro
- OS: [e.g. iOS8.1] iOS 17.0 Simulator

**Additional context**

**Comment by Teakowa** [comment]
It may require some logical changes, but based on the current maintenance status of this repository, I suggest temporarily locking the version to bypass the error.

```diff
- .package(url: "https://github.com/daltoniam/Starscream.git", from: "4.0.0"),
+ .package(url: "https://github.com/daltoniam/Starscream.git", .exact("4.0.4")),
```

You can use my fork: https://github.com/Teakowa/swift-graphql

**Comment by shaps80** [comment]
Also resolved in #164 depending on which is merged. However I agree we should lock the version to avoid this in future releases.

**Comment by shaps80** [comment]
This was resolved earlier this week #164 👍
